Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -65,150 +61,158 @@ class SerializerTest : public ::testing::Test {
   Serializer serializer;
 
   void ExpectRoundTrip(const FieldValue& model,
-                       const std::vector<uint8_t>& bytes,
+                       const google::firestore::v1beta1::Value& proto,
                        FieldValue::Type type) {
+    // First, serialize model with our (nanopb based) serializer, then
+    // deserialize the resulting bytes with libprotobuf and ensure the result is
+    // the same as the expected proto.
+    ExpectSerializationRoundTrip(model, proto, type);
+
+    // Next, serialize proto with libprotobuf, then deserialize the resulting
+    // bytes with our (nanopb based) deserializer and ensure the result is the
+    // same as the expected model.
+    ExpectDeserializationRoundTrip(model, proto, type);
+  }
+
+  google::firestore::v1beta1::Value ValueProto() {
```

Comment:
Maybe? It's slightly awkward, since nullptr_t is supposed to be for null *pointers*, whereas this is more of a null *value*. But we can't take in a null value, since the compiler will get NULL confused with 0. On the whole, I like this better. Done.